lib/eventsource.dart

```diff
@@ -55,46 +55,89 @@ class EventSource extends Stream<Event> {
   EventSourceDecoder _decoder;
   String _body;
   String _method;
-
+  final _openOnlyOnFirstListener;
+  final _closeOnLastListener;
+  http.ByteStream responseStream;
 
   /// Create a new EventSource by connecting to the specified url.
   static Future<EventSource> connect(url,
-      {http.Client client, String lastEventId, Map headers, String body, String method}) async {
+      {http.Client client,
+      String lastEventId,
+      Map headers,
+      String body,
+      String method,
+      bool openOnlyOnFirstListener,
+      bool closeOnLastListener}) async {
     // parameter initialization
     url = url is Uri ? url : Uri.parse(url);
     client = client ?? new http.Client();
     lastEventId = lastEventId ?? "";
     body = body ?? "";
     method = method ?? "GET";
-    EventSource es = new EventSource._internal(url, client, lastEventId, headers, body, method);
-    await es._start();
+    EventSource es = new EventSource._internal(url, client, lastEventId,
+        headers, body, method, openOnlyOnFirstListener, closeOnLastListener);
+    if (!es._openOnlyOnFirstListener) {
+      await es._start();
+    }
     return es;
   }
 
-  EventSource._internal(this.url, this.client, this._lastEventId, this.headers, this._body, this._method) {
+  EventSource._internal(
+      this.url,
+      this.client,
+      this._lastEventId,
+      this.headers,
+      this._body,
+      this._method,
+      bool openOnlyOnFirstStream,
+      bool closeOnLastStreamClosing)
+      : _openOnlyOnFirstListener = openOnlyOnFirstStream ?? false,
+        _closeOnLastListener = closeOnLastStreamClosing ?? false {
+    // initialize here so we can close the stream
+    _streamController = new StreamController<Event>.broadcast(
+        onCancel: () => _lastStreamDisconnected());
+
     _decoder = new EventSourceDecoder(retryIndicator: _updateRetryDelay);
   }
 
   // proxy the listen call to the controller's listen call
   @override
   StreamSubscription<Event> listen(void onData(Event event),
-          {Function onError, void onDone(), bool cancelOnError}) =>
-      _streamController.stream.listen(onData,
-          onError: onError, onDone: onDone, cancelOnError: cancelOnError);
+      {Function onError, void onDone(), bool cancelOnError}) {
+    if (_readyState == EventSourceReadyState.CLOSED &&
+        _openOnlyOnFirstListener) {
```

Comment:
Is it dartfmt making this into 2 lines like this?

Reply:
Because personally I prefer it to be one line.

Reply:
yes, dartfmt did it.